demos/device-shadow-for-aws-iot-embedded-sdk/shadow_demo_main.c

```diff
@@ -0,0 +1,700 @@
+         */
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            /* Desired power on state . */
+            LogInfo( ( "Send desired power state with 1.\n\n" ) );
+
+            ( void ) memset( pcUpdateDocument,
+                             0x00,
+                             sizeof( pcUpdateDocument ) );
+
+            snprintf( pcUpdateDocument,
+                      SHADOW_DESIRED_JSON_LENGTH + 1,
+                      SHADOW_DESIRED_JSON,
+                      ( int ) 1,
+                      ( long unsigned ) ( xTaskGetTickCount() % 1000000 ) );
+
+            returnStatus = PublishToTopic( SHADOW_TOPIC_STRING_UPDATE( THING_NAME ),
+                                           SHADOW_TOPIC_LENGTH_UPDATE( THING_NAME_LENGTH ),
+                                           pcUpdateDocument,
+                                           ( SHADOW_DESIRED_JSON_LENGTH + 1 ) );
+        }
+
+        if( returnStatus == EXIT_SUCCESS )
+        {
+            /* Note that PublishToTopic already called MQTT_ProcessLoop,
+             * therefore responses may have been received and the prvEventCallback
+             * may have been called, which may have changed the stateChanged flag.
+             * Check if the state change flag has been modified or not. If it's modified,
+             * then we publish reported state to update topic.
+             */
+            if( stateChanged == true )
+            {
+                /* Report the latest power state back to device shadow. */
+                LogInfo( ( "Report to the state change: %d\n\n", ulCurrentPowerOnState ) );
+                ( void ) memset( pcUpdateDocument,
+                                 0x00,
+                                 sizeof( pcUpdateDocument ) );
+
+                /* Keep the client token in global variable used to compare if
+                 * the same token in /update/accepted. */
+                ulClientToken = ( xTaskGetTickCount() % 1000000 );
+
+                snprintf( pcUpdateDocument,
+                          SHADOW_REPORTED_JSON_LENGTH + 1,
+                          SHADOW_REPORTED_JSON,
+                          ( int ) ulCurrentPowerOnState,
+                          ( long unsigned ) ulClientToken );
+
+                returnStatus = PublishToTopic( SHADOW_TOPIC_STRING_UPDATE( THING_NAME ),
+                                               SHADOW_TOPIC_LENGTH_UPDATE( THING_NAME_LENGTH ),
+                                               pcUpdateDocument,
+                                               ( SHADOW_DESIRED_JSON_LENGTH + 1 ) );
+            }
+            else
+            {
+                LogInfo( ( "No change from /update/delta, unsubscribe all shadow topics and disconnect from MQTT.\r\n" ) );
+            }
+        }
+
+        LogInfo( ( "Start to unsubscribe shadow topics and disconnect from MQTT. \r\n" ) );
+
+        if( EXIT_FAILURE == UnsubscribeFromTopic( SHADOW_TOPIC_STRING_UPDATE_DELTA( THING_NAME ),
```

Comment:
How come we don't assign returnStatus to these failures? I think failure to unsubscribe should make the demo fail?